contrib/infiniband/ibvctx.c

```diff
@@ -1113,21 +1113,32 @@ int _get_cq_event(struct ibv_comp_channel * channel,
   int rslt, flags;
 
   internal_channel = ibv_comp_to_internal(channel);
-  flags = fcntl(internal_channel->real_channel->fd, F_GETFL, NULL);
+  flags = NEXT_FNC(fcntl)(internal_channel->real_channel->fd,
+                          F_GETFL, NULL);
 
   // We need to change the call to non-blocking mode
-  if (flags & O_NONBLOCK == 0) {
+  // FIXME: we need to take care of the case where fd is already
+  // changed to non-blocking mode by the user.
+  if (!(flags & O_NONBLOCK)) {
+    rslt = NEXT_FNC(fcntl)(internal_channel->real_channel->fd,
+                 F_SETFL, flags | O_NONBLOCK);
```

Comment:
Could you please fix the indentation here? Align it with the first argument on the previous line.